tests/unit/amo/components/TestFeaturedAddons.js

```diff
@@ -6,59 +6,284 @@ import {
+      },
+      loading: false,
+      results: null,
+    });
+
+    sinon.assert.calledWith(dispatch, _getFeatured({
+      addonType,
+      errorHandlerId: customErrorHandler.id,
+    }));
+  });
+
+  it('fetches featured add-ons when component is updated', () => {
+    const firstAddonType = ADDON_TYPE_EXTENSION;
+    const secondAddonType = ADDON_TYPE_THEME;
+    const dispatch = sinon.stub();
+
+    const root = render({
+      dispatch,
+      params: {
+        visibleAddonType: VISIBLE_ADDON_TYPES_MAPPING[firstAddonType],
+      },
+      loading: false,
+      results: null,
+    });
+
+    dispatch.reset();
+
+    root.setProps({
+      params: {
+        visibleAddonType: VISIBLE_ADDON_TYPES_MAPPING[secondAddonType],
+      },
+    });
+
+    sinon.assert.calledWith(dispatch,
+      _getFeatured({ addonType: secondAddonType }));
+  });
+
+  it('fetches featured add-ons when addonType changes', () => {
+    const firstAddonType = ADDON_TYPE_EXTENSION;
+    const secondAddonType = ADDON_TYPE_THEME;
+    const dispatch = sinon.stub();
+
+    store.dispatch(_loadFeatured({ addonType: firstAddonType }));
+    // Get the results props from the state mapper.
+    const mappedProps = mapStateToProps(store.getState());
+
+    const root = render({
+      dispatch,
+      params: {
+        visibleAddonType: VISIBLE_ADDON_TYPES_MAPPING[firstAddonType],
+      },
+      ...mappedProps,
+    });
+
+    dispatch.reset();
+
+    root.setProps({
+      params: {
+        visibleAddonType: VISIBLE_ADDON_TYPES_MAPPING[secondAddonType],
+      },
+    });
+
+    sinon.assert.calledWith(dispatch,
+      _getFeatured({ addonType: secondAddonType }));
+  });
+
+  it('does not fetch featured add-ons while component is loading', () => {
+    const addonType = ADDON_TYPE_EXTENSION;
+    const dispatch = sinon.stub();
+
+    const root = render({
+      dispatch,
+      params: {
+        visibleAddonType: VISIBLE_ADDON_TYPES_MAPPING[addonType],
+      },
+      loading: false,
+      results: null,
+    });
+
+    dispatch.reset();
+
+    root.setProps({
+      loading: true,
+      results: null,
+    });
+
+    sinon.assert.notCalled(dispatch);
+  });
+
+  it('does not fetch featured add-ons when results are loaded', () => {
+    const addonType = ADDON_TYPE_EXTENSION;
+    const dispatch = sinon.stub();
+
+    store.dispatch(_loadFeatured());
+    // Get the results props from the state mapper.
+    const mappedProps = mapStateToProps(store.getState());
+
+    const root = render({
+      dispatch,
+      params: {
+        visibleAddonType: VISIBLE_ADDON_TYPES_MAPPING[addonType],
+      },
+      ...mappedProps,
+      hasSearchParams: true,
+    });
+
+    dispatch.reset();
+
+    root.setProps({
+      // Update an unrelated parameter.
+      hasSearchParams: false,
+    });
+
+    sinon.assert.notCalled(dispatch);
+  });
+
+  it('does not fetch featured add-ons when results are loaded + empty', () => {
```

Comment:
Just a note that these test descriptors are all really helpful and awesome–they describe each state very clearly and make reviewing these tests a breeze. Thanks! 👍